project1/application.py

```diff
@@ -107,12 +109,67 @@ def auth():
 
 @app.route("/logout")
 def logout():
-    session.pop('Username', None)
+    session.pop("username", None)
     return redirect(url_for('index'))
 
 
-@app.route("/home/<Username>")
-def userHome(Username):
-    if Username in session:
-        return render_template("userDetails.html", username=Username, message="Successfully logged in.", heading="Welcome back")
-    return redirect(url_for('index'))
+
+@app.route("/search",methods=["POST","GET"])
+def search():
+	if "username" in session:
+		username = session["username"]
+
+		if request.method == "GET":
+			return render_template("search.html")
+		else:
+			result = request.form.get("search")
+			result = '%'+result+'%'
+			search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result))).all()
+			if len(search_result) == 0:
+				return render_template("search.html",message = "No records found")
+			return render_template("search.html", books=search_result)
+	else:
+		return redirect(url_for("register"))
+
+
+@app.route("/bookpage", methods=["GET"])
+def bookpage():
+	if request.method == "GET":
+		return render_template("bookpage.html")
+
+
+@app.route("/api/search/", methods=["POST"])
+def api_search():
+	if request.method == "POST":
+		jsobj = request.json
+
+		result = jsobj["search"]
+		result = '%' + result + '%'
+		search_result = Books.query.filter(or_(Books.tittle.ilike(result), Books.author.ilike(result), Books.isbn.ilike(result))).all()
+		if result is None:
```

Comment:
Other error/exceptions like db connection failure need to be handled.

Reply:
Handled.